### PR TITLE
Mvc

### PR DESCRIPTION
**Pull request by mikevh17**
## Purpose
Assets folder: alle assets zoals css, js en afbeeldingen

Config folder: db connection file, mogelijk extra bestand voor hashed gegevens.

Controllers folder: Alle controllers die we gaan gebruiken komen hier in. Naamgeving: *naam*Controller.php. Bestaat uit ÉÉN class met als naam de bestandsnaam - .php, minstens één functie (met de naam index!). Meer uitleg zie MVC - guide, drive -> sprint 1.

Core folder: super classes voor controllers en models. Functies die in de subclasses gebruikt kunnen worden komen hier.

Models folder: Alle models die we gaan gebruiken komen hier in. Naamgeving: *naam*Model.php. Bestaat uit ÉÉN class met als naam de bestandsnaam - .php. Hier komen functies in die gegevens uit de database halen.

Views folder: Alle views die we gaan gebruiken komen hier in. Naamgeving: duidelijke naam. Bestanden omvatten HTML, JS en PHP.

Webroot folder: Bevat de hoofd index.php en .htaccess. Hoeft in principe niets aan gewijzigd te worden.

Dispatcher.php file: Wordt aangeroepen in webroot/index.php. Laad de juiste controller met de juiste url.

Request.php file: Geeft de url terug.

Router.php file: Hakt de url in stukken en maakt deze beschikbaar voor dispatcher.php.

## Outcome
Efficiënter aan userstories werken. Betere bestand scheiding. MVC leren en begrijpen.

## Details
Basis code voor mvc designpatern.

### CODE REVIEW

File: Config/db.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    public static function getBdd()
+    {
+        $hostname = "(local)";
+        $dbname = "";
+        $username = "";
+        $pw = "";
+
+        if (is_null(self::$bdd)) {
+            self::$bdd = new PDO("mysql:host=localhost;dbname=deb107033n3_test", 'deb107033n3_test', 'abc123');
+            //new PDO ("sqlsrv:Server=$hostname;Database=$dbname;ConnectionPooling=0", "$username", "$pw");
```
Comment (Stijnwingens) [inline Config/db.php]: Could you explain this please?

File: Controllers/productController.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+class productController extends Controller
+{
+    //public = kan gezien worden op site, private = kan niet gezien worden op site en kan hier gebruikt worden.
+    function index()
+    {
```
Comment (Stijnwingens) [inline Controllers/productController.php]: Try to keep your code styling consistent. Here you open the parenthesis on the next line. 5 lines down you open on the same line. Please fix your styling before merging. I prefer:

`function functionName(){`
`code here;`
`}`

But whatever you choose now decides the styling for the rest of the project, so please remain consistent.

File: Controllers/productController.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    function view($par = null) {
+        print_R($_GET['test']);
+//        echo $par;
+        $data['page'] = "product";
+        $this->set($data);
+        $this->load_view("template");
+    }
+
+    private function show() {
```
Comment (Stijnwingens) [inline Controllers/productController.php]: What is shown? Try to keep naming clear to other developers. After some digging i understood this function shows a product. Maybe a naming like "productController.showProduct()" would be more suited here?

I know this seems nitpicky, but if you start doing it like this from the start things will stay clean, and will be easier to maintain.

File: Core/Controller.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+class Controller
+{
+    public function __construct()
+    {
+    }
+
+    var $vars = [];
+
+    function set($data)
```
Comment (Stijnwingens) [inline Core/Controller.php]: What is set? Please explain.

File: Views/Includes/head.inc.php [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<head>
+    <title>Eenmaal Andermaal Veilingen</title>
```
Comment (Stijnwingens) [inline Views/Includes/head.inc.php]: The title is not the same for every page, so this needs to be set with a PHP variable. Please add this to the head file before merging.

File: Views/home.php [hunk 6]
```diff
@@ -0,0 +1,111 @@
+<div class="uk-card uk-card-default uk-card-body uk-width-1-5 uk-position-center-left">
```
Comment (Stijnwingens) [inline Views/home.php]: Please remove this file as a more recent version will soon be added via the homepage branch.

File: Webroot/index.php [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<?php
```
Comment (Stijnwingens) [inline Webroot/index.php]: Could you explain in a bit more details what happens here?

File: assets/css/uikit-rtl.css [hunk 8]
```diff
@@ -0,0 +1,12184 @@
+/*! UIkit 3.1.3 | http://www.getuikit.com | (c) 2014 - 2018 YOOtheme | MIT License */
```
Comment (Stijnwingens) [inline assets/css/uikit-rtl.css]: In the head we still use the UIKit JS files. Do we still need them when we have all the CSS files? And are the CSS files already implemented? If so, where?

File: request.php [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<?php
+    class Request
+    {
+        public $url;
+
+        public function __construct()
+        {
+            $this->url = $_SERVER["REQUEST_URI"];
```
Comment (Stijnwingens) [inline request.php]: REQUEST_URI? Shouldn't it be REQUEST_URL?

Comment (mikevh17) [reply request.php]: No, there is a difference between URI and URL. When it is changed to REQUEST_URL it doesn't work.

File: Webroot/index.php (hunk 7 above)
Comment (mikevh17) [reply Webroot/index.php]: This index.php file is the 'head' file and gets directed to by the .htaccess files. The file adds the required files and eventually 'forms' the page.

File: Core/Controller.php (hunk 4 above)
Comment (mikevh17) [reply Core/Controller.php]: The set($parameter) function sends the parameter to the core/controller.php file and makes it global for used files so they can be used in the view.

File: Controllers/productController.php (hunk 2 above)
Comment (mikevh17) [reply Controllers/productController.php]: My tidy function did this. Normally use the mentioned style. Changed

File: Controllers/productController.php (hunk 3 above)
Comment (mikevh17) [reply Controllers/productController.php]: Naming should be short as the class and file name already contain the main subject. Longer names also means that they are longer in an url e.g. http://www.example.com/product/show/10, http://www.example.com/product/productController_ShowProduct/10.

File: Config/db.php (hunk 1 above)
Comment (mikevh17) [reply Config/db.php]: It creates the connection with the database. The lines between the if statement have to be changed so it fits the server database credentials.

File: Views/Includes/head.inc.php (hunk 5 above)
Comment (mikevh17) [reply Views/Includes/head.inc.php]: Added a parameter that can be used for this.

File: assets/css/uikit-rtl.css (hunk 8 above)
Comment (mikevh17) [reply assets/css/uikit-rtl.css]: Some functions (like the autofiill) use the JS files. All css and js files are in the assets folder. They are implemented in the head (views/includes/head.php).

File: Views/home.php (hunk 6 above)
Comment (mikevh17) [reply Views/home.php]: Removed

File: request.php (hunk 9 above)
Comment (Stijnwingens) [reply request.php]: Okay 👍

**Pull request merged**